scripts/dts/edtlib.py

```diff
@@ -890,6 +890,23 @@ def __repr__(self):
             "binding " + self.binding_path if self.binding_path
                 else "no binding")
 
+    def _init_status(self):
+        # Initializes Node._node_status.
+        #
+        # Node._node_status holds node enabled condition. This is dynamically
+        # evaluated by the node status property and If his parent status is
+        # equal to 'disabled' the node became 'disabled' too. This ensures that
+        # the whole tree branch is disabled.
+        self._node_status = "status" not in self._node.props or \
+            self._node.props["status"].to_string() != "disabled"
+
+        parent = self.parent
+        while parent:
+            if not parent.enabled:
+                self._node_status = False
+                return
+            parent = parent.parent
```

Comment:
The `for dt_node in self._dt.node_iter():` loop visits parent nodes before child nodes, so `self.parent.enabled` is already initialized here. Might work to replace `_init_status()` with this:

```python
parent_enabled = not self.parent or parent.enabled
self_enabled = "status" not in self._node.props or \
    self._node.props["status"] != "disabled"
return parent_enabled and self_enabled
```

Reply:
Yes, no problem!